VersionInfo.cpp

```diff
@@ -181,15 +185,176 @@ static void fill_compiler_info(unichar* s) {
 
 
 #ifdef __DATE__
+// see https://stackoverflow.com/questions/11697820/how-to-use-date-and-time-predefined-macros-in-as-two-integers-then-stri
```

Comment:
What about this:

1. Use only `UNITEX_VERSION_BUILD_DATE` on `VersionInfo.cpp` no extra logic is needed

The, you have several alternatives, for exemple:

2a. Customize your `Makefile` to redefine `UNITEX_VERSION_BUILD_DATE` as explained here: https://stackoverflow.com/a/31336873/204287, or

2b. Create a `VersionLocal.h` header redefining `UNITEX_VERSION_BUILD_DATE`

```cpp
// process __DATE__ here
#undef UNITEX_VERSION_BUILD_DATE
#define UNITEX_VERSION_BUILD_DATE BUILD_YEAR_CH0 BUILD_YEAR_CH1 ...
```
3. Then include conditionally `VersionLocal.h` on `VersionInfo.h`

```cpp
...
#include "Version.h"
...
#if (defined(UNITEX_VERSION_BUILD_IS_ANONYMOUS) &&\
             UNITEX_VERSION_BUILD_IS_ANONYMOUS)
# include "VersionLocal.h"
#endif
...
```